.github/workflows/main.yml

```diff
@@ -12,41 +12,40 @@ on:
 
 jobs:
   build-and-test:
-    name: Test PY=${{ matrix.python-version }}, DEV=${{ matrix.developer }}, EXP=${{ matrix.experimental }}, DEP=${{ matrix.deprecated }}
-    runs-on: ubuntu-latest
+    name: Test PY=${{ matrix.python-version }}, BCD=${{ matrix.bitcoind-version }}, EXP=${{ matrix.experimental }}, DEP=${{ matrix.deprecated }}
+    runs-on: ubuntu-22.04
     timeout-minutes: 60
     strategy:
       fail-fast: false
       matrix:
-        python-version: [3.7, 3.8, 3.9]
-        developer: [0,1]
+        python-version: ["3.8", "3.9", "3.10", "3.11", "3.12"]
+        bitcoind-version: ["22.0", "26.0"]
         experimental: [1]
         deprecated: [0]
 
     steps:
-    - uses: actions/checkout@v3
+    - uses: actions/checkout@v4
 
-    - name: Checkout c-lightning@master
-      uses: actions/checkout@v3
+    - name: Checkout c-lightning@v23.11
+      uses: actions/checkout@v4
       with:
         repository: 'ElementsProject/lightning'
         path: 'lightning'
-        ref: 'master'
+        ref: 'v23.11'
```

Comment:
I'm not sure if this is a good idea, testing against master will make the CI fail faster when stuff breaks, so things will break more often, but we'll know sooner.

Thoughts @cdecker ?